app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -423,13 +423,17 @@ const User = RestModel.extend({
     return ajax(userPath(`${this.username_lower}/preferences/email`), {
       type: "POST",
       data: { email },
+    }).then(() => {
+      this.unconfirmed_emails.pushObject(email);
```

Comment:
These two fix a bug where adding or changing an email would add it to the list until refresh